File: Arrays/reverseArray.cpp
#include <iostream>
using namespace std;

void swap(int *a, int *b)
{
    int temp = *a;
    *a = *b;
    *b = temp;
}

void reverseArray(int *array, int size)
{
    int x = 0, y = size - 1;

    while (x < y)
    {
        swap(array[x], array[y]);

        x++;
        y--;
    }
}

int main()
{
    int array[] = {1, 55, 66, 43, 26, 94};
    int size = (sizeof array) / (sizeof array[0]);

    for (int i = 0; i < size; i++)
    {
        cout << array[i] << " ";
    }

    cout << endl;
    reverseArray(array, size);

    for (int i = 0; i < size; i++)
    {
        cout << array[i] << " ";
    }

    return 0;
}

File: Arrays/reverse_array.py
def reverse_array(array):
    x = 0
    y = len(array) - 1

    while x < y:
        temp = array[x]
        array[x] = array[y]
        array[y] = temp
        x += 1
        y -= 1

if __name__ == '__main__':
    array = [1, 2, 3, 4, 5]

    print('Original array:')
    for i in array:
        print(i)

    reverse_array(array)
    
    print('Reversed array:')
    for i in array:
        print(i)